src/firefly/js/charts/ui/PinnedChartPanel.jsx

```diff
@@ -139,7 +149,32 @@ export const PinChart = ({viewerId, tbl_group}) => {
     return canPin ? <TextButton onClick={doPinChart} title='Pin the active chart'>Pin Chart</TextButton> : null;
 };
 
-export function pinChart({chartId, autoLayout=true}) {
+export function pinChart({chartId, autoLayout=true }) {
+
+    // there are cases when chart is not fully loaded.  if so, wait before pinning the chart
+    if (isChartLoading(chartId)) {
+        dispatchAddActionWatcher({actions:[CHART_UPDATE], callback: (action, cancelSelf) => {
+                const aChartId = action.payload?.chartId;
+                if (chartId === aChartId) {
+                    if (!isChartLoading(chartId)) { //chart is fully loaded now
+                        cancelSelf?.();
+                        doPinChart({chartId, autoLayout});
+                    }
+                } else {
+                    // fallback to ensure there's no orphan watchers
+                    cancelSelf?.();
+                }
+            }
+        });
+    }
+    else {
+        doPinChart({chartId,autoLayout});
+    }
+}
```

Comment:
This function is a little hard to read it does down 4 levels. It would be easier if you just started it with:
```js
if (!isChartLoading(chartId)) {
      doPinChart({chartId,autoLayout});
      return;
}
```
and the not `else`

In the same way inside of call back function, start with (no `else`):
```js
if (chartId !== aChartId) {
  cancelSelf?.()
  return;
}
```
The whole thing will be less deep and easier to follow.